gemma-core/src/main/java/ubic/gemma/core/datastructure/matrix/BaseExpressionDataMatrix.java

```diff
@@ -129,6 +130,14 @@ public BioAssayDimension getBestBioAssayDimension() {
 
     }
 
+    @Override
+    public Collection<Integer> findRowsByName( String name ) {
+        return rowElementMap.entrySet().stream()
+                .filter( entry -> entry.getKey().getName().equals( name ) )
+                .map( entry -> entry.getValue() )
+                .collect( Collectors.toList() );
```

Comment:
Here, you can do `mapToInt` and `toArray` to return a `int[]`. It would be more efficient to use than a collection of integer objects. Then `getRows` would take int an array instead.

Reply:
I don't think that's important. Typically for debugging (the purpose of this) only one gene would be sought.